platforms/core-configuration/model-core/src/main/java/org/gradle/api/internal/model/NamedObjectInstantiator.java

```diff
@@ -98,7 +96,7 @@ public <T> T fromState(Class<T> type, Object state) {
 
     public <T extends Named> T named(final Class<T> type, final String name) throws ObjectInstantiationException {
         try {
-            return type.cast(generatedTypes.get(type, cacheFactory).getUnchecked(name));
+            return type.cast(generatedTypes.get(type, this::cacheFactory).getUnchecked(name));
```

Comment:
🤔 this one is the most suspicious to me in terms of potential performance impact. Some quick research suggests that capturing lambdas are not cached, so we're going to have at least an object allocation every time we hit this callsite. Maybe this is a premature optimisation, of course.

Reply:
I won't take that risk :).
I'll add a suppression for this.
I guess there is a good chance this was the reason for prior implementation.